src/core/hle/D3D8/XbPixelShader.cpp

```diff
@@ -1485,6 +1512,17 @@ void DxbxUpdateActivePixelShader() // NOPATCH
 	  }
 	  fColor[PSH_XBOX_CONSTANT_FRONTFACE_FACTOR].r = frontfaceFactor;
 
+	  fogEnable = XboxRenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) > 0;
```

Comment:
This reads a float, does a check turning it into a float and then assigns the result to a float - but it's not very clear what float value that is.

Perhaps it's good to make that more explicit somehow, perhaps using a ternary expression setting between two constants (FLOAT_TRUE vs FLOAT_FALSE for instance)?

Reply:
changed to  
`fogEnable = (NV2ARenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) > 0 == Fog_Enabled) ? Fog_Enabled : Fog_Disabled;` and 
`fogEnable = (XboxRenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) > 0 == Fog_Enabled) ? Fog_Enabled : Fog_Disabled;`
 and added  
 float Fog_Enabled = 1;
 float Fog_Disabled = 0;
 Verified it does work

Reply:
GetXboxRenderState() returns a DWORD not a float. Patrick might mistake it as GetXboxRenderStateAsFloat(). So the original code should work correctly. but it would be more feasible to use 
  fogEnable = XboxRenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) != 0;
I would suggest to use the code above or 
  fogEnable = XboxRenderStates.GetXboxRenderState(xbox::_X_D3DRENDERSTATETYPE::X_D3DRS_FOGENABLE) != false;